### PR TITLE
🐛 Remove old state in IG on update

### LINKED ISSUE #110: Old state in IG json should be removed before validation
**Scenario**
1. User adds resource file to IG: `StructureDefinition-participant.json`
2. User runs validation and it passes
3. User changes resource id from `participant` to `foo`, and file name to `StructureDefinition-participant.json`
4. User runs validation and it fails because IG publisher can't find StructureDefinition.id=participant

The old configuration for `StructureDefinition.id=participant` needs to be removed from the IG json. 

**Solution**
Any entries in the IG JSON must have a corresponding resource file in the input being supplied to IG publisher. If an entry does not have a corresponding resource file then it should be removed.

### PR DESCRIPTION
**Pull request by znatty22**
Closes #110 and #114

### CODE REVIEW

File: kf_model_fhir/app.py [hunk 1]
```diff
@@ -333,10 +343,9 @@ def _update_ig_config(
             resources_dict[entry['reference']['reference']] = entry
         else:
             del rd[entry['reference']['reference']]
-
-        if rm_file:
-            os.rmfile(rd['filepath'])
-            logger.info(f'🗑 Deleted resource file {rd["filepath"]}')
+            if rm_file:
```
Comment (znatty22) [inline kf_model_fhir/app.py:346]: FYI - This change is not related to the issues addressed in this PR. Just noticed that this blob should have been indented. Didn't want to create a whole separate PR for it.

File: README.md [hunk 2]
```diff
@@ -77,8 +77,14 @@ resources subdirectory: `./site_root/input/resources`.
 
 3. Validate the models
 
-   `fhirmodel validate ./site_root/ig.ini --publisher_opts='-tx n/a'`
+   ```shell
+   # Method 1 - Uses the dockerized IG publisher
+   fhirmodel validate ./site_root/ig.ini --publisher_opts='-tx n/a'
 
+   # Method 2 - Uses native IG publisher - faster than above method
+   fhirmodel add ./site_root/input/resources
```
Comment (liberaliscomputing) [inline README.md:85]: Previously, it didn't recursively look into all sub-directories/-files. Does it now? If not, we may want to describe that.
